### ISSUE TITLE
Proximity flow not responding

### ISSUE THREAD

**Issue by maurolucc** [eu-digital-identity-wallet/eudi-app-ios-wallet-ui#197]
Hello!

**Context**: I'm trying to test the [Proximity flow](https://github.com/eu-digital-identity-wallet/eudi-app-ios-wallet-ui?tab=readme-ov-file#proximity-flow) following the instructions provided. Once the wallet is successfully build on an iPhone, I load the sample data provided and click on show QR code.

**Problem**: The EUDI Verifier app detects the QR code and shows "Device engagement received..." message. After a while, it shows a toast "Error connecting holder" and nothing happens in the wallet. I've seen in the demo video that this flow works on Eudi app Android wallet. Is it currently working fot Eudi app iOS wallet?

Versions tested:

- Wallet: code built at this [commit](cff9e9fdaa1fe6497aaf11b52e51223fd3dbe60b)
- Verifier: [version 2.4](https://install.appcenter.ms/orgs/eu-digital-identity-wallet/apps/mdoc-verifier-testing/distribution_groups/eudi%20verifier%20(testing)%20public)
- Verifier options combinations:  BLE central client mode, BLE peripheral server mode, BLE central client mode + BLE peripheral server mode.

Thanks in advance

**Comment by maurolucc** [comment]
Hello Stilianos,

What do you mean but develop branch? I only see main branch in the repository.

I've just pulled the latest version of this repository in main branch, I've compiled the app and test the proximity flow with the App verifier and I still can't make it work:

![image](https://github.com/user-attachments/assets/82b1c243-5d4d-4749-8391-01828bf997d6)
![image](https://github.com/user-attachments/assets/2af97073-da01-49c9-bf18-e997d8f7a11a)
![image](https://github.com/user-attachments/assets/def4ea21-67a6-4d60-866f-448c051125da)
![image](https://github.com/user-attachments/assets/77fed2d5-305e-4219-8414-fab54d379fb1)

I've tried different combinations of App Verifier settings but still doesn't work.

I've seen in the README of this repository that the demo of the proximity flown shows the Android EUDI wallet (not the iOS EUDI wallet) performing the proximity flow with the App Verifier.  Is the iOS EUDI wallet ready to perform a proximity flow?

**Comment by stzouvaras** [comment]
Hi @maurolucc,

Yes, I meant the main branch. The iOS app works and supports proximity, and it functions without any issues on my end.

If it doesn't work for you, there might be an issue with the verifier app, your device model, or its settings. According to this [documentation](https://github.com/eu-digital-identity-wallet/eudi-app-ios-wallet-ui/blob/main/wiki/verifier_proximity.md), the proximity verifier app is not exclusively developed by us, and support is currently limited until a new proximity verifier is ready.

My test scenario was:

- Download the proximity verifier without touching any settings and simply providing necessary permissions (Android API level is important here) Galaxy Z Fold 6
- Run iOS EUDI main branch iPhone 11
- run the proximity presentation flow.

**Comment by maurolucc** [comment]
Thanks for your quick answer.

I've reproduced your scenario with a Galaxy Note8 for the verifier and an iPhone mini 12 for the wallet and although all the permissions are granted, I obtain the same result unfortunately. 

I will keep investigating...

**Comment by stzouvaras** [comment]
Maybe try restarting your devices and give it another shot. I’ve noticed that sometimes the connection can be a bit finicky, depending on the device accessing the BLE module.

**Comment by maurolucc** [comment]
Hello! 

Today I had the chance to test the flow with several iPhones. These are the results (iOS version; Bluetooth Modem Firmware version):

- **iPhone XR** _(18.0.1; 7.00.00)_: **OK** ✅
- **iPhone 13** _(18.0.1; 4.02.05)_: **KO** ❌
- **iPhone 12** _(17.6.1; 4.70.01)_: **KO** ❌
- **iPhone 12** _(17.6; 4.70.00)_: **KO** ❌
- **iPhone 12 mini** _(17.6.1; 4.70.01)_: **KO** ❌

Only one of them worked although all of them have Bluetooth enabled. It looks like the modem firmware version might be the problem.

Which iPhone have you used to test?

**Comment by stzouvaras** [comment]
Hi @maurolucc ,

iPhone 11 18.0.1 modem version: 6.00.00
iPhone 15 pro max 18.0.1 and iOS 17.+ (I don't have the modem version info)

**Comment by stzouvaras** [comment]
Hi @maurolucc, if you’ve identified that the issue is related to the BT modem and compatibility, please let me know so I can forward the issue to the appropriate team.

**Comment by maurolucc** [comment]
Indeed, from my experiments, it looks that BT modem versions has an impact. It is early to have a conclusion about this issue but it deserves further investigation. All what I could discover has been already shared in this thread.
